pkg/provider/provider.go

```diff
@@ -222,3 +281,55 @@ func ParsePrivateKey(privateKeyPath string) (*rsa.PrivateKey, error) {
 	}
 	return rsaPrivateKey, nil
 }
+
+type Result struct {
+	AccessToken string `json:"access_token"`
+	TokenType   string `json:"token_type"`
+	ExpiresIn   int    `json:"expires_in"`
+}
+
+func GetOauthData(refreshToken, redirectUrl string) url.Values {
+	data := url.Values{}
+	data.Set("grant_type", "refresh_token")
+	data.Set("refresh_token", refreshToken)
+	data.Set("redirect_uri", redirectUrl)
+	return data
+}
+
+func GetOauthRequest(dataContent *strings.Reader, endPoint, clientId, clientSecret string) *http.Request {
+	request, err := http.NewRequest("POST", endPoint, dataContent)
+	if err != nil {
+		errors.Wrap(err, "Request to the endpoint could not be completed")
+	}
+	request.SetBasicAuth(clientId, clientSecret)
+	request.Header.Set("Content-Type", "application/x-www-form-urlencoded;charset=UTF-8")
+	return request
+
+}
+
+func GetOauthAccessToken(
+	endPoint,
+	client_id,
+	client_secret string,
+	data url.Values) (string, error) {
+
+	client := &http.Client{}
+	request := GetOauthRequest(strings.NewReader(data.Encode()), endPoint, client_id, client_secret)
+
+	var result Result
+
+	response, err := client.Do(request)
+	if err != nil {
```

Comment:
```suggestion
	if err {
```

Reply:
I think we want to keep `if err != nil` but should return if `err != nil`